push-to-s3-and-invalidate-cloudfront/action.yml

```
@@ -0,0 +1,150 @@
+    required: false
+    default: ""
+  s3-dir-to-upload-to:
+    description: |
+      The S3 directory to upload the files to. If unspecified, the default `PR-$pull-request-number` will be used
+      or the directory may be infered from the github context. A value of `/` indicates the root of the bucket.
+    required: false
+    default: "pr-previews"
+  s3-dir-to-delete-from:
+    description: |
+      The directory on S3 whose contents need to be deleted. If unspecified, the default `PR-$pull-request-number` will be used
+      or the directory may be infered from the github context
+    required: false
+    default: "pr-previews"
+  s3-files-to-exclude:
+    description: |
+      A pattern or a comma separated list of files to exclude when running the aws s3 sync command. 
+      Refer to https://docs.aws.amazon.com/cli/latest/reference/s3/index.html#use-of-exclude-and-include-filters or more information
+      Set this variable to an empty string if you don't want to exlcude any files
+      If unspecified will default to `pr-previews/*`
+    required: false
+    default: "pr-previews/*"
+  invalidate-cloudfront-cache:
+    description: A flag to indicate if the cloudfront cache should be invalidated
+    required: false
+    default: "false"
+
+outputs:
+  dir_to_upload_to:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.upload.dir_to_upload_to }}
+  dir_to_delete_from:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.delete.dir_to_delete_from }}
+  files_to_exlude:
+    description: Files to exclude from S3 when using the sync command.
+    value: ${{ steps.exclude.files_to_exclude }}
+
+runs:
+  using: composite
+  steps:
+    - name: Compute the S3 directory to upload to
+      shell: bash
+      id: upload
+      run: |
+        if [[ "${{ inputs.s3-dir-to-upload-to }}" == "pr-previews" ]]
+        then
+          dir_to_upload_to="${{ inputs.s3-dir-to-upload-to }}/PR-${{ inputs.pull-request-number }}"
+        elif [[ "${{ inputs.s3-dir-to-upload-to }}" == "/" ]]
+        then
+          dir_to_upload_to=""
+        else
+          dir_to_upload_to="${{ inputs.s3-dir-to-upload-to }}"
+        fi
+        echo "dir_to_upload_to=$dir_to_upload_to" >> $GITHUB_OUTPUT
+
+    - name: Compute the S3 directory to delete from
+      shell: bash
+      id: delete
+      run: |
+        if [[ "${{ inputs.s3-dir-to-delete-from }}" == "pr-previews" ]]
+        then
+          dir_to_delete_from="${{ inputs.s3-dir-to-delete-from }}/PR-${{ inputs.pull-request-number }}"
+        elif [[ "${{ inputs.s3-dir-to-delete-from }}" == "/" ]]
+        then
+          dir_to_delete_from=""
+        else
+          dir_to_delete_from="${{ inputs.s3-dir-to-delete-from }}"
+        fi
+        echo "dir_to_delete_from=$dir_to_delete_from" >> $GITHUB_OUTPUT
```

Can we combine both these into one step? I am not seeing a scenario where dir_to_upload_to and dir_to_delete_from are different values from within the same workflow call. And both these code blocks are identical:

```suggestion
    - name: Compute the S3 directory to interact with
      shell: bash
      id: s3_directory
      run: |
        if [[ "${{ inputs.s3-dir }}" == "pr-previews" ]]
        then
          dir_name="${{ inputs.s3-dir }}/PR-${{ inputs.pull-request-number }}"
        elif [[ "${{ inputs.s3-dir }}" == "/" ]]
        then
          dir_name=""
        else
          dir_name="${{ inputs.s3-dir }}"
        fi
        echo "dir_name=$dir_name" >> $GITHUB_OUTPUT
```
So instead of having two variables "s3-dir-to-upload-to" and "s3-dir-to-delete-from". You have one input "s3-dir" which is the directory the workflow cares about and depending on the the context and value passed to "s3-action" you either upload or delete from that directory.

---

Done!